packages/flutter_tools/lib/src/localizations/localizations_utils.dart

```diff
@@ -294,7 +294,14 @@ String generateString(String value) {
 /// Given a list of strings, placeholders, or helper function calls, concatenate
 /// them into one expression to be returned.
 ///
-/// If `isSingleStringVar` is passed, then we want to convert "'$expr'" to "expr".
+/// Each of the strings in [expressions] should be a raw string, which, if it were to be added to a dart file,
+/// would be a properly formatted dart string with escapes and/or interpolation. The purpose of this function is to
+/// concatenate these dart strings into a single dart string which can be returned.
+///
+/// The following rules describe all the cases which needs to be handled:
+/// 1. String with only a single interpolated variable is converted to the variable itself e.g. ["'$expr'"] -> "expr".
+/// 2. If one string ends in an interpolation and the next begins with an alphanumeric character, then the former should
```

Comment:
Shouldn't the rules mention an empty string and the reverse of `#2`: 
```dart
["'another string'", "'$expr1'"] -> "'another string'${expr1}".
```

Reply:
It is acceptable to not put braces in the reverse case since `"'another string$expr1'"` is a valid Dart string. But I'll add a mention of what happens when `expressions` is empty.